flink-runtime/src/main/java/org/apache/flink/runtime/execution/Environment.java

```diff
@@ -157,6 +158,13 @@
 
 	GlobalAggregateManager getGlobalAggregateManager();
 
+	/**
+	 * Get the enabled external resource drivers for external resources.
+	 *
+	 * @return the enabled external resource drivers for external resources
+	 */
+	Map<String, ExternalResourceDriver> getExternalResourceDrivers();
```

Comment:
I would prefer to not expose the `ExternalResourceDriver` to all `Environment` users if possible. Wouldn't it be enough to expose `Set<ExternalResourceInfo> getExternalResourceInfo(String resourceName)`?

Reply:
Or maybe the `Environment` allows you to get an `ExternalResourceInfoProvider/Lookup`.

Reply:
I think it would be good enough to expose `Set<ExternalResourceInfo> getExternalResourceInfo(String resourceName)`

Reply:
I think `Map<String, ExternalResourceInfo> getExternalResourceInfos` would be good enough for current version.